packages/workspace/src/browser/workspace-frontend-contribution.ts

```diff
@@ -97,6 +104,40 @@ export class WorkspaceFrontendContribution implements CommandContribution, Keybi
             new UriAwareCommandHandler(this.selectionService, {
                 execute: (uri: URI) => this.saveAs(uri),
             }));
+        commands.registerCommand(WorkspaceCommands.UPLOAD, new FileSelection.CommandHandler(this.selectionService, {
+            multi: false,
+            isEnabled: selection => this.isEnabled(selection),
+            isVisible: selection => this.isVisible(selection),
+            execute: selection => this.upload(selection)
+        }));
+    }
+
+    protected isEnabled(selection: FileSelection | undefined): boolean {
+        return !environment.electron.is();
+    }
+
+    protected isVisible(selection: FileSelection | undefined): boolean {
+        return !environment.electron.is() && (selection === undefined || selection.fileStat.isDirectory);
+    }
+
+    protected async upload(selection: FileSelection): Promise<void> {
+        try {
+            if (selection === undefined) {
+                await this.uploadService.upload(this.workspaceService.tryGetRoots()[0].uri);
```

Comment:
Since we have multi-root workspace support enabled #6089, I think we should use the `quickPick` to prompt users to select a root to upload to. Also `tryGetRoots()[0].uri` will throw an exception if no roots are present.

Reply:
We should ask only, if there is more than one root.